dotCMS/src/main/java/com/dotmarketing/portlets/rules/parameter/ParameterDefinition.java

```diff
@@ -58,5 +64,67 @@ public int getPriority() {
     public void checkValid(ParameterModel model) throws InvalidRuleParameterException, RuleEngineException {
     	this.inputType.checkValid(model.getValue());
     }
-}
 
+    /**
+     * Utility type used to correctly read immutable object from JSON representation.
+     *
+     * @deprecated Do not use this type directly, it exists only for the <em>Jackson</em>-binding
+     * infrastructure
+     */
+    @Deprecated
+    @JsonDeserialize
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
+    @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.NONE)
+    static final class Json {
+
+        @javax.annotation.Nullable
+        String key;
+        @javax.annotation.Nullable
+        String i18nBaseKey;
+        @javax.annotation.Nullable
+        String defaultValue;
+        @javax.annotation.Nullable
+        Input inputType;
+        @javax.annotation.Nullable
+        int priority;
+
+        @JsonProperty("key")
+        public void setKey(@Nullable final String key) {
+            this.key = key;
+        }
+
+        @JsonProperty("i18nBaseKey")
+        public void setI18nBaseKey(@Nullable final String i18nBaseKey) {
+            this.i18nBaseKey = i18nBaseKey;
+        }
+
+        @JsonProperty("defaultValue")
+        public void setDefaultValue(@Nullable final String defaultValue) {
+            this.defaultValue = defaultValue;
+        }
+
+        @JsonProperty("inputType")
+        public void setInputType(@Nullable Input inputType) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'inputType' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4925354)